custom_components/tesla_custom/__init__.py

```diff
@@ -162,27 +221,16 @@ async def async_setup_entry(hass, config_entry):
     for device in all_devices:
         entry_data["devices"][device.hass_type].append(device)
 
-    for platform in PLATFORMS:
-        _LOGGER.debug("Loading %s", platform)
-        hass.async_create_task(
-            hass.config_entries.async_forward_entry_setup(config_entry, platform)
-        )
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
+
     return True
 
 
 async def async_unload_entry(hass, config_entry) -> bool:
     """Unload a config entry."""
-    unload_ok = all(
-        await asyncio.gather(
-            *[
-                hass.config_entries.async_forward_entry_unload(config_entry, platform)
-                for platform in PLATFORMS
-            ]
-        )
+    unload_ok = await hass.config_entries.async_unload_platforms(
+        config_entry, PLATFORMS
     )
-    await hass.data[DOMAIN].get(config_entry.entry_id)[
```

Comment:
The controller should get the disconnect signal if we're going down so it stops polling Tesla.